websocket/websocket-server.go

```diff
@@ -0,0 +1,335 @@
+package websocket
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+
+	runtime "github.com/Hammad887/chat-app"
+	domain "github.com/Hammad887/chat-app/models"
+)
+
+// ctx is a variable of type context.Context, which is used to manage request-scoped values, deadlines, and cancellations.
+var ctx context.Context
+
+// upgrader use for upgrading http connection to websocket connection
+var upgrader = websocket.Upgrader{
+	// ReadBufferSize specifies the size of the buffer used for reading from the WebSocket connection.
+	ReadBufferSize: 1024,
+
+	// WriteBufferSize specifies the size of the buffer used for writing to the WebSocket connection.
+	WriteBufferSize: 1024,
+
+	// EnableCompression allows for message compression when set to true.
+	EnableCompression: true,
+
+	// CheckOrigin is a function that checks the origin of the WebSocket request to prevent cross-origin attacks.
+	// In this case, it's set to always return true, meaning any origin is accepted.
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
```

Comment:
Please remove the whole  logic from your server. In an ideal situation, microservices are not publicly exposed directly to the internet. Instead, we typically use a server-client proxy (e.g., Nginx) that serves as the public-facing entry point to your server

Reply:
Removed